src/main/java/org/snomed/snowstorm/rest/MultiSearchController.java

```diff
@@ -75,7 +78,71 @@ public ItemsPage<BrowserDescriptionSearchResult> findDescriptions(
 				.conceptActive(conceptActive);
 
 		PageRequest pageRequest = ControllerHelper.getPageRequest(offset, limit);
-		Page<Description> descriptions = multiSearchService.findDescriptions(descriptionCriteria, pageRequest);
+		Page<Description> descriptions = multiSearchService.findDescriptions(descriptionCriteria, ecl, pageRequest, null);
+		timer.checkpoint("description search");
+
+		Map<String, List<Description>> branchDescriptions = new HashMap<>();
+		Map<String, List<String>> branchConceptIds = new HashMap<>();
+		for (Description description : descriptions) {
+			branchDescriptions.computeIfAbsent(description.getPath(), s -> new ArrayList<>()).add(description);
+			branchConceptIds.computeIfAbsent(description.getPath(), s -> new ArrayList<>()).add(description.getConceptId());
+		}
+
+		List<LanguageDialect> languageDialects = ControllerHelper.parseAcceptLanguageHeaderWithDefaultFallback(acceptLanguageHeader);
+		Map<String, ConceptMini> conceptMiniMap = new HashMap<>();
+		for (String branchPath : branchConceptIds.keySet()) {
+			conceptMiniMap.putAll(conceptService.findConceptMinis(branchPath, branchConceptIds.get(branchPath), languageDialects).getResultsMap());
+			timer.checkpoint("Join concepts from " + branchPath);
+		}
+
+		List<BrowserDescriptionSearchResult> results = new ArrayList<>();
+		for (Description description : descriptions) {
+			BrowserDescriptionSearchResult result = new BrowserDescriptionSearchResult(description.getTerm(), description.isActive(), description.getLanguageCode(),
+					description.getModuleId(), conceptMiniMap.get(description.getConceptId()));
+			result.addExtraField("branchPath", description.getPath());
+			results.add(result);
+		}
+		timer.finish();
+
+		return new ItemsPage<>(new PageImpl<>(results, pageRequest, descriptions.getTotalElements()));
+	}
+	
+	@Operation(description = "Search descriptions across multiple Code Systems. Can search specified branches in addition to all published branches.")
+	@PostMapping(value = "multisearch/descriptions", produces = {"application/json", "text/csv"})
+	@JsonView(value = View.Component.class)
```

Comment:
@RichardAWood, I see there is a conversation going on about the validity of the Post method. If the Post method is being kept and the only difference is to support multiple branches, then perhaps the logic in the Get method and Post method should be extracted into a private method. This would help ensure both endpoints behave the same way.

From what I can see, the only difference is:

`multiSearchService.findDescriptions(descriptionCriteria, ecl, pageRequest, null);`

vs

`multiSearchService.findDescriptions(descriptionCriteria, ecl, pageRequest, includeBranches.getBranches());`

Reply:
That makes sense to me, as long as everyone else is on board.  @CoderMChu and @pgwilliams, thoughts?

Reply:
Yes it makes sense.

Reply:
findDescriptionsHelper method has been created, and is called by both Get and Post methods.